OneCommunityGlobal/HGNRest
# Sucheta - Make It Known When New AI Prompt is Available

# Description
(PRIORITY HIGH) Jae: Make it clear when a new AI Prompt is available (WIP Sucheta)
- Related [PR1510](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1510)
- The AI Prompt is now editable. I’d like it to be obvious to a person when they see the button below for the first time after the AI Prompt has been updated.
<img width="599" alt="Screenshot 2024-01-20 at 8 41 37 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/101291283/24234f66-de03-41a4-bea3-07fef4f25752">

- Ideally, I’d like it to look like what’s shown below: 
   - Cleaned up [200x160px image link ](https://www.dropbox.com/scl/fi/0g9cvqlzy9gj00sie28mk/New-HGN-Icon-11kb-200x160px.png?rlkey=vkngel8ezl0cbhbjutmlotvin&dl=0)to use. Give me the exact dimensions needed and I’ll create a new one. 
- After a person has copied the link for the first time, the button should return to what you see above.
<img width="711" alt="Screenshot 2024-01-20 at 8 39 50 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/101291283/bd90bb5c-27a7-4c0e-9e98-f887405a8a52">


## Related PRS (if any):

To test this backend PR you need to checkout the [#1899](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1899) frontend PR.
…

## Main changes explained:
- Adds new property `modifiedDateTime` to `weeklySummaryAIPrompt` - model
- Adds new property `copiedAiPrompt` to `userProfile` - model
- Adds new end points to get and put the `CopiedPromptDate`.
- Updates the `modifiedDateTime` when the owner edits the Ai prompts.
- Updates the `copiedAiPrompt` when a user copies the Ai prompt.
…

## How to test:
1. Check into current branch
2. Do `npm install` , `nmp run build` and `npm start`  to run this PR locally
3. Clear site data/cache from browser's `Application` tab.
 
Test as `OWNER`: 
1. Log as `Owner` user
2. Go to WeeklySummary submission modal, then click the button `View and Copy Current AI Prompt`. As an owner, the user should have the permission to edit the AI prmpt. 
3. Once the AI prmpt has been updated, the button changes to show a `new` icon - making it visually known that the AI prompt has been updated. 
4. The visual indication will only persist until the user has copied the updated Ai prompt. Once copied , the button changes and the `new` icon disappers.

Test as any other user:

1. Log as any user
2. Go to WeeklySummary submission modal. 
3. The button `View and Copy Current Ai Prompt` should be normal if there is no change in the AI prompt since last time the prompt was copied
4. If the AI prmpt was updated, the button changes to show a `new` icon - making it visually known that the AI prompt has been updated. 
5. The visual indication will only persist until the user has copied the updated Ai prompt. Once copied , the button changes and the `new` icon disappers.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/101291283/12ab0141-28a5-41ba-afc5-133aa4f9b71a




## Note:
Include the information the reviewers need to know.
